src/Splunk.Logging.Common/FixedSizeQueue.cs

```diff
@@ -23,19 +24,22 @@ namespace Splunk.Logging
     /// and a new item is queued, the oldest item in the queue is dropped.
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    internal class FixedSizeQueue<T> : ConcurrentQueue<T>
+    internal class FixedSizeQueue<T>
     {
         public int Size { get; private set; }
         public IProgress<bool> Progress = new Progress<bool>();
         public bool IsCompleted { get; private set; }
 
+        private readonly BlockingCollection<T> _collection = new BlockingCollection<T>();
```

Comment:
It might be good to add a comment saying the default backing queue is `ConcurrentQueue`, which means it is FIFO (useful for the dropping logic below).